test/tests/functional/pbs_reliable_job_startup.py

```diff
@@ -5435,3 +5454,214 @@ def test_t20(self):
                                       "3gb", 3, 3,
                                       self.jobA_place,
                                       self.jobA_sel_esc)
+
+    @timeout(400)
+    def test_t21(self):
+        """
+        Test: radio silent moms causing the primary mom to not get
+              any acks from the sister moms executing prologue hooks.
+              After some 'job_launch_delay' time has passed, primary
+              mom will consider node hosts that have not acknowledged
+              the prologue hook execution as failed hosts, and will
+              not use their vnodes in the pruning of jobs.
+        """
+        job_launch_delay = 120
+        c = {'$job_launch_delay': job_launch_delay}
+        self.momA.add_config(c)
+
+        # instantiate queuejob hook
+        hook_event = "queuejob"
+        hook_name = "qjob"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.qjob_hook_body)
+
+        # instantiate execjob_prologue hook
+        hook_event = "execjob_prologue"
+        hook_name = "prolo"
+        a = {'event': hook_event, 'enabled': 'true', 'alarm': 60}
+        self.server.create_import_hook(hook_name, a, self.prolo_hook_body5)
+
+        # instantiate execjob_launch hook
+        hook_event = "execjob_launch"
+        hook_name = "launch"
+        a = {'event': hook_event, 'enabled': 'true'}
+        self.server.create_import_hook(hook_name, a, self.launch_hook_body)
+
+        # First, turn off scheduling
+        a = {'scheduling': 'false'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        jid = self.create_and_submit_job('job1')
+        # Job gets queued and reflects the incremented values from queuejob
+        # hook
+        self.server.expect(JOB, {'job_state': 'Q',
+                                 'tolerate_node_failures': 'job_start',
+                                 'Resource_List.mem': '10gb',
+                                 'Resource_List.ncpus': 13,
+                                 'Resource_List.nodect': 5,
+                                 'Resource_List.select': self.job1_iselect,
+                                 'Resource_List.site': self.job1_oselect,
+                                 'Resource_List.place': self.job1_place,
+                                 'schedselect': self.job1_ischedselect},
+                           id=jid, attrop=PTL_AND)
```

Comment:
It's op, not attrop.

Reply:
No it's 'attrop'. I looked into test/fw/ptl/lib/pbs_testlib.py, and under class Server, def expect(), it showed:
def expect(self, obj_type, attrib=None, id=None, op=EQ, attrop=PTL_OR,
               attempt=0, max_attempts=None, interval=None, count=None,
               extend=None, offset=0, runas=None, level=logging.INFO,
               msg=None):
with the here document:
  :param attrop: Operation on multiple attributes, either
                       PTL_AND, PTL_OR when an PTL_AND is used, only
                       batch objects having all matches are
                       returned, otherwise an OR is applied